Introduction

rswetnam opened the pull request:
In this proposed revision, I am trying to concentrate on the broader picture of the xAPI is designed to be used and have eliminated the paragraphs describing how the interoperability works. I think that is explained below.

garemoko commented on xAPI.md:
```diff
@@ -260,49 +260,30 @@ learning that SCORM could not enable.
 
 <a name="partone"/>
-#Part One: About The Experience API
+#PART ONE: LEARNING EXPERIENCES
```
I don't like all caps for readability and accessibility. 

I also prefer the original title. This part is about the xAPI, its not about learning experiences.

rswetnam replied:
The caps detnoting the main parts of the document is a convention I've seen used elsewhere to show larger divisions - however I don't feel strongly about it and if you think it detracts from readibility - I will change.

I don't like the original title - as it makes it sound like a sidebar - however I will resubmit with original title and propose a different title in another pull request where I hope it makes sense.

garemoko commented on xAPI.md:
```diff
@@ -260,49 +260,30 @@ learning that SCORM could not enable.
-#Part One: About The Experience API
+#PART ONE: LEARNING EXPERIENCES
 
+<a name="Introduction"/>
```
Take care changing anchors. It looks like you didn't change this in the TOC. Anchors should be all lower case and 'introduction' is not a good anchor as there may be multiple introductions across the documents/sections. Maybe ```partone-introduction```.

rswetnam replied:
Thanks - good point - I will change.  How would you feel about a convention that uses the full name of the Section title in the tag - lower case and using underscores rather than dashes?  I found that doing this made it much easier to navigate the document in my text editor as I could click on part_one_introduction and get the whole phrase where as if there were dashes, the editor would just select whatever word I was on.

garemoko replied:
+1 to a convention. I like my way better than your way. <-(This is probably the heart of all standardisation problems.)

garemoko commented on xAPI.md:
```diff
@@ -260,49 +260,30 @@ learning that SCORM could not enable.
+#PART ONE: LEARNING EXPERIENCES
 
+<a name="Introduction"/>
+## 1.0 Introduction
```
+1 to this.

garemoko commented on xAPI.md:
```diff
@@ -260,49 +260,30 @@ learning that SCORM could not enable.
 
-<a name="roleofxapi"/>
-## 1.0 Summary of the Experience API
+The Experience API (xAPI) is a technical specification that aims to facilitate the documentation and communication of learning experiences.  It does so by setting out a protocol that specifies how learning experiences should be described and an Application Programming Interface that sets out how information about these experiences should be stored and exchanged electronically.  
```
Suggest splitting this second sentence in two as 'setting out an API that sets out' is not great. 

The specification does not set up how anything should be stored, just how it can be sent and retrieved.

rswetnam replied:
Yes, good point - when I was referring to storage, I had in mind broader concepts such as statements being immutable and they should not be changed when being stored rather than details of implementation such as using a relational or non-relational database.  I will rephrase.

garemoko replied:
To be really pedantic, you can change statements as much as you like whilst they are stored, so long as you return them in their original state.

garemoko commented on xAPI.md:
```diff
@@ -260,49 +260,30 @@ learning that SCORM could not enable.
-facilitates communication about experiences between a Learning Record Store and a client, 
-typically an Activity Provider. This specification was developed with learning experiences in mind, 
-but many other kinds of experiences could be tracked using xAPI. 
+The xAPI is part of the Advanced Distributed Learning (ADL) initiative, established in 1997 to standardize and modernize training and education management and delivery in the US. Among the many outcomes of this initiative was SCORM - the Shareable Content Object Reference Model. SCORM has 
```
This seems helpful. Perhaps "overseen by" or something instead of "part of"

andyjohnson replied:
I would expand on US to either US DoD or US Government.  That was the mission, but luckily the work went so much further!

garemoko commented on xAPI.md:
```diff
@@ -260,49 +260,30 @@ learning that SCORM could not enable.
-typically an Activity Provider. This specification was developed with learning experiences in mind, 
-but many other kinds of experiences could be tracked using xAPI. 
+The xAPI is part of the Advanced Distributed Learning (ADL) initiative, established in 1997 to standardize and modernize training and education management and delivery in the US. Among the many outcomes of this initiative was SCORM - the Shareable Content Object Reference Model. SCORM has 
+become a great success and an internationally recongized framework that has greatly contributed greatly to the economy and effectiveness of online learning.
```
"great success" that has "greatly contributed greatly". "has been a great success". That's a lot of greatness. 

How about "SCORM has become an internationally recongized framework ubiquitously adopted by Learning Management Systems and authoring tools alike."

You can then keep the "great" in the next paragraph.

rswetnam replied:
Great point!

garemoko commented on xAPI.md:
```diff
@@ -260,49 +260,30 @@ learning that SCORM could not enable.
+The xAPI is part of the Advanced Distributed Learning (ADL) initiative, established in 1997 to standardize and modernize training and education management and delivery in the US. Among the many outcomes of this initiative was SCORM - the Shareable Content Object Reference Model. SCORM has 
+become a great success and an internationally recongized framework that has greatly contributed greatly to the economy and effectiveness of online learning.
+  
+While SCORM has been a great success, there has been a growing realization that there is a huge need and potential to track learning experiences of individuals beyond formal, structured computer-based traing.  In assessing candidates' suitability for positions or their capability form performing various tasks, there is a need to consider a wide range of formal and informal learning experiences, b ...
```
"computer-based traing" -> "computer based training"

rswetnam replied:
I will try to spell beter.  However, I believe that computer-based is a compound-modifier (see e.g. https://en.wikipedia.org/wiki/Compound_modifier) and should be spelled with a hyphen.

garemoko replied:
Maaaybe, but CBT is a thing so maybe it should all be hyphened?

rswetnam replied:
no - its an acronym

On Wed, Sep 23, 2015 at 11:11 AM, Andrew Downes <notifications@github.com>
wrote:

> In xAPI.md
> <https://github.com/adlnet/xAPI-Spec/pull/759#discussion_r40237173>:
>
> >
> > -The Experience API (xAPI) is an Application Programming Interface (API) specification that
> > -facilitates communication about experiences between a Learning Record Store and a client,
> > -typically an Activity Provider. This specification was developed with learning experiences in mind,
> > -but many other kinds of experiences could be tracked using xAPI.
> > +The xAPI is part of the Advanced Distributed Learning (ADL) initiative, established in 1997 to standardize and modernize training and education management and delivery in the US. Among the many outcomes of this initiative was SCORM - the Shareable Content Object Reference Model. SCORM has
> > +become a great success and an internationally recongized framework that has greatly contributed greatly to the economy and effectiveness of online learning.
> > +
> > +While SCORM has been a great success, there has been a growing realization that there is a huge need and potential to track learning experiences of individuals beyond formal, structured computer-based traing.  In assessing candidates' suitability for positions or their capability form performing various tasks, there is a need to consider a wide range of formal and informal learning experiences, both on and offline.  That information, more often than not be scattered across a wide variety of sources.
>
> Maaaybe, but CBT is a thing so maybe it should all be hyphened?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/adlnet/xAPI-Spec/pull/759/files#r40237173>.
>



-- 
Roger Swetnam
p.  604-408-5703
c.  778-848-3118

garemoko replied:
I just Googled "computer based training" and all the websites use "computer-based training". I stand corrected.

garemoko commented:
Thanks for this @rswetnam! I'm adding some specific comments. One general note is that there are a lot of double spaces between sentences which can cause problems when the markdown is rendered.

garemoko commented on xAPI.md:
```diff
@@ -260,49 +260,30 @@ learning that SCORM could not enable.
-This specification does not attempt to standardize how the Activity Provider facilitates and tracks the learning experience, 
-nor how the Learning Store stores and procesess that data. The scope of this specification is limited to the requests and responses
-between Learning Record Store and client. 
+Out of this perceived need, the xAPI initiative and specification was born.  xAPI assumes that:
```
"xAPI initiative" - > "xAPI Working Group"

garemoko replied:
This comment needs to be addressed. You also still have a lot of double spaces. I'll re-review this once that's done.

rswetnam replied:
I use the term "xAPI initiative" as a broad term to include the formation of the xAPI Working Group and the the undertaking of work on this document - the xAPI Specification as well as a lot of other activities such as the plugfests connected with the xAPI.  I confess to not understanding the problem here.  I have removed the double spaces between sentences.

garemoko replied:
Sorry, I should have explained more. To me "xAPI Initiative" sounds like an organization, but no such organization exists so i think it could be confusing. We could just delete "initiative and "? 

I guess really you're referring to a continuum made up of Project Tin Can and the xAPI Working Group.

rswetnam replied:
For me, putting the term initiative in lower case suggests that we are not
talking about an organization.  I would like to leave initiative in because
a whole bunch more that the specification was was born from this need - not
the least of which is the communities of practice.

On Wed, Sep 30, 2015 at 11:11 AM, Andrew Downes <notifications@github.com>
wrote:

> In xAPI.md
> <https://github.com/adlnet/xAPI-Spec/pull/759#discussion_r40829424>:
>
> >
> > -This specification does not attempt to standardize how the Activity Provider facilitates and tracks the learning experience,
> > -nor how the Learning Store stores and procesess that data. The scope of this specification is limited to the requests and responses
> > -between Learning Record Store and client.
> > +Out of this perceived need, the xAPI initiative and specification was born.  xAPI assumes that:
>
> Sorry, I should have explained more. To me "xAPI Initiative" sounds like
> an organization, but no such organization exists so i think it could be
> confusing. We could just delete "initiative and "?
>
> I guess really you're referring to a continuum made up of Project Tin Can
> and the xAPI Working Group.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/adlnet/xAPI-Spec/pull/759/files#r40829424>.
>



-- 
Roger Swetnam
p.  604-408-5703
c.  778-848-3118

garemoko commented on xAPI.md:
```diff
@@ -260,49 +260,30 @@ learning that SCORM could not enable.
-are some examples of additional technologies which the Experience API is designed
-to support. While the implementation details of these services are not specified here,
-the Experience API is designed with this larger architectural vision in mind.
+The xAPI is designed pursue these goals by setting out a common framework by which organizations and systems can gather, store and exchange data about learning experiences. 
```
"designed pursue" -> "pursues"

Gathering and storing data are out of scope of xAPI. This is the point that some of the text you've cut is trying to make clear.

garemoko commented on xAPI.md:
```diff
@@ -260,49 +260,30 @@ learning that SCORM could not enable.
+The goals of the xAPI are:
+
+* To make it easier to understand and compare learning experiences recorded in a wide variety of places.
+* To maximize interoperability of systems which create, gather, store and process information about learning experiences.
```
gathering and storing are fine here, because we're talking about interoperability between systems that do this, not setting out how these things are done.

garemoko commented on xAPI.md:
```diff
@@ -260,49 +260,30 @@ learning that SCORM could not enable.
+* To provide a guide to those who want to build applications that conform to and implement this specification.
 * To provide criteria against which conformance to this specification can be tested.
 
-There are two main ways that the xAPI promotes interoperability between systems that 
```
I think these three paragraphs contain useful content that should be included somewhere, not removed.

andyjohnson replied:
Agreed.  It may need to be a subsection as I think once we are out of 1.0 it isn't going to fit.

garemoko commented on xAPI.md:
```diff
@@ -260,49 +260,30 @@ learning that SCORM could not enable.
 
-The Experience API (xAPI) is an Application Programming Interface (API) specification that 
-facilitates communication about experiences between a Learning Record Store and a client, 
-typically an Activity Provider. This specification was developed with learning experiences in mind, 
```
I know @aaronesilvers and others were keen to keep this sentence in.

rswetnam replied:
I am keen to remove this sentence.  Using terms of art such as Learning Store or Activity Provider in the Introduction before they have been defined in the document is disrespectful to the reader and suggests that the document is aimed at an old boys club whose members already know what the secret handshakes mean.  The introduction should concentrate on the broader issues of intent and goals.

garemoko replied:
Sorry, I meant the sentence starting "This specification was developed with learning experiences in mind,"

andyjohnson commented on xAPI.md:
```diff
@@ -260,49 +260,30 @@ learning that SCORM could not enable.
-nor how the Learning Store stores and procesess that data. The scope of this specification is limited to the requests and responses
-between Learning Record Store and client. 
+Out of this perceived need, the xAPI initiative and specification was born.  xAPI assumes that:
+  * There is a need to be able to analyze information about learning experiences distributed across a wide variety of sources.
```
I'd say not only sources, but platforms and/or technologies.

garemoko replied:
Agree.

andyjohnson commented:
"initiative and specification was born" should be "were born" - I think this is looking pretty good.

garemoko commented:
Just a note that there's a number of comments and discussions on outdated diffs (sadly github hides these by default) that need to be resolved before this is merged. Might be worth running through those on the next call.

garemoko commented:
Still to do/discuss:
* https://github.com/adlnet/xAPI-Spec/pull/759#discussion_r40225465
* https://github.com/adlnet/xAPI-Spec/pull/759#discussion-diff-40225746
* https://github.com/adlnet/xAPI-Spec/pull/759#discussion_r40226719
* https://github.com/adlnet/xAPI-Spec/pull/759#discussion-diff-40227669 (this is a big one)
* https://github.com/adlnet/xAPI-Spec/pull/759#discussion-diff-40228602 (also strong feelings on this)

andyjohnson commented:
Per the 10/21/15 call, need to get rid of the "big brother" tracking notions, stick to learning (which includes outcomes)

garemoko commented:
TODO:
1. Change _ to - in anchors
2. Re-write the second sentence on line 268 as discussed [here](https://github.com/adlnet/xAPI-Spec/pull/759#discussion_r40225746).
3. Add some wording to make clear that xAPI is about learning **and the outcome of learning**, not just pure learning

That's all.

rswetnam commented:
Changes:

1. removed underscore in anchors
2. rewrote second sentence in line 268
3. Used learning and learning outcomes in two sentences in the introduction

garemoko commented:
Ok. I think we're good to go! +1

andyjohnson commented:
+1 here too, merging
